src/common/debug.c

```diff
@@ -238,12 +237,19 @@ daos_key2str(daos_key_t *key)
 				break;
 			}
 		}
-		if (can_print) {
+
+		if (can_print)
 			strncpy(buf, key->iov_buf, len);
-			buf[len] = 0;
-		} else {
+		else
 			strcpy(buf, "????");
+
+		len = min(len, strlen(buf));
+		if (key->iov_len == sizeof(uint64_t)) {
+			sprintf(buf + len, " uint64:"DF_U64, *(uint64_t *)key->iov_buf);
+			len += strlen(buf + len);
 		}
+		D_ASSERTF(len < (DF_KEY_STR_SIZE - 1), "len %d\n", len);
```

Comment:
This seems like a bad idea as this is written.  If the key is already close to DF_KEY_STR_SIZE, then the uint64:DF_U64 will put it over the top and trigger this assertion.

I think we need to have both a MAX_STR_SIZE and a MAX_SIZE that accounts for the extra characters.  It's ok to assert if it exceeds that.

Reply:
>This seems like a bad idea as this is written. If the key is already close to DF_KEY_STR_SIZE, then the uint64:DF_U64 will put it over the top and trigger this assertion.

Hmm, "uint64:DF_U64" will only be appended if iov_len == 8, so it will be far less than MAX_STR_SIZE.

Reply:
Ah, you're right.   Sorry for the noise.